watertap/property_models/multicomp_aq_sol_prop_pack.py

```diff
@@ -1186,8 +1227,133 @@ def rule_visc_k_phase(b, p):
+                "for these components becuase the HaydukLadie method is selected.".format(
+                    common_ind
+                )
+            )  # warning for components whose diffusivity_data will be overriden by the HaydukLaudie method.
+            self.diffus_phase_comp = Var(
+                self.params.phase_list,
+                self.mv_dt_ind | self.diffus_dt_ind,
+                initialize=1e-9,
+                units=pyunits.m**2 * pyunits.s**-1,
+                doc="mass diffusivity of solute components",
+            )
+            self.hl_diffus_cont = Param(
+                mutable=True,
+                default=13.26e-9,
+                initialize=13.26e-9,
+                units=pyunits.dimensionless,
+                doc="Hayduk Laudie correlation constant",
+            )
+            self.hl_visc_coeff = Param(
+                mutable=True,
+                default=1.14,
+                initialize=1.14,
+                units=pyunits.dimensionless,
+                doc="Hayduk Laudie viscosity coefficient",
+            )
+            self.hl_molar_volume_coeff = Param(
+                mutable=True,
+                default=0.589,
+                initialize=0.589,
+                units=pyunits.dimensionless,
+                doc="Hayduk Laudie molar volume coefficient",
+            )
+
+        def rule_diffus_phase_comp(b, p, j):
+            if self.params.config.diffus_calculation == DiffusivityCalculation.none:
+                b.diffus_phase_comp[p, j].fix(self.params.config.diffusivity_data[p, j])
+                return Constraint.Skip
+            elif (
+                self.params.config.diffus_calculation
+                == DiffusivityCalculation.HaydukLaudie
+            ):
+
+                if j not in self.mv_dt_ind:
+                    b.diffus_phase_comp[p, j].fix(
+                        self.params.config.diffusivity_data[p, j]
+                    )
+                    return Constraint.Skip
+                else:
+
+                    diffus_coeff_inv_units = pyunits.s * pyunits.m**-2
+                    visc_solvent_inv_units = pyunits.cP**-1
+                    molar_volume_inv_units = pyunits.mol * pyunits.cm**-3
+                    return (b.diffus_phase_comp[p, j] * diffus_coeff_inv_units) * (
+                        (
+                            pyunits.convert(b.visc_d_phase[p], to_units=pyunits.cP)
+                            * visc_solvent_inv_units
+                        )
+                        ** b.hl_visc_coeff
+                    ) * (
+                        (
+                            pyunits.convert(
+                                b.molar_volume_comp[p, j],
+                                to_units=pyunits.cm**3 * pyunits.mol**-1,
+                            )
+                            * molar_volume_inv_units
+                        )
+                        ** b.hl_molar_volume_coeff
+                    ) == b.hl_diffus_cont
+
```

Comment:
@lbibl I did not use `value()` when the H-L equation was implemented in the unit model because it ran the risk of incorrect calculations when `units_meta` was not as expected. However, the property packages have hardcoded units which the unit models directly adopt, so this doesn't risk the issue. Still, I think this method has less ambiguity for what units are correct for the parameters used in the equation, and holds true if for some reason the units defined at the var declaration are ever changed.